FEATURE: Add sorting to list

suffle opened the pull request:
This PR adds sorting to the list view. It reimplements nearly the same feature as in the old media browser.

Closes: #28

suffle commented on Classes/GraphQL/Resolver/Type/QueryResolver.php:
```diff
@@ -174,7 +179,17 @@ protected function createAssetProxyQuery(
 
-        // TODO: Implement sorting via `SupportsSortingInterface`
+        if ($sortBy && $assetProxyRepository instanceof SupportsSortingInterface) {
+            switch ($sortBy) {
```
This implementation equals the one from the old media browser. I was thinking of a more flexible solution, e.g. go for `$assetProxyRepository->orderBy([$sortBy => $sortDirection]);` and leave the sortB completely to the frontend, but decided a more explicit version here would be better

Sebobo replied:
Yes that's better 👍

suffle commented on Resources/Private/JavaScript/media-module/src/components/TopBar/TopBar.tsx:
```diff
@@ -24,7 +25,10 @@ const TopBar: React.FC = () => {
 
-    const components = useMemo(() => [ClipboardActions, SearchBox, TypeFilter, ViewModeSelector], []);
+    const components = useMemo(
+        () => [ClipboardActions, SearchBox, TypeFilter, SortOrderSelector, ViewModeSelector],
```
Hopefully the positioning is OK here, I wasn't sure, if there are any designs / mockups on what goes where

Sebobo replied:
Yes that's fine, there are no official guidelines yet. I mostly used the ideas collected in the discuss topics about the media management.

Sebobo commented on Resources/Private/JavaScript/core/src/hooks/useAssetsQuery.ts:
```diff
@@ -14,7 +14,7 @@ import {
 } from '../state';
 import { ASSETS } from '../queries';
-
+import selectedSortOrderState, { SORT_BY, SORT_DIRECTION } from '../state/selectedSortOrderState';
```
You can combine the import into the line 14 above with the state variables.

suffle replied:
So importing the selectedSortOrderState with line 14 and the enums separately? Oder do you propose exporting the enums through the state/index.ts as well? Or maybe there is an even better place where to put them?

Sebobo commented:
Thx ❤️
Will test this in the next days. Looks good so far by reading!

I would probably also like to have this persisted to keep the order when leaving the module and coming back later. But I encountered some issues with my initial implementation of persisted view options while implementing #115.
So this can be done later separately.

Sebobo commented:
@suffle the select box creates some issues when used in the UI in smaller screens:

<img width="1228" alt="Bildschirmfoto 2022-03-04 um 10 08 33" src="https://user-images.githubusercontent.com/596967/156735891-0592306a-4a18-43d5-a1a5-d641bf2e02a6.png">

We should probably only show the icon when the space is limited and `selectionMode` is set. Probably the same when `isInNodeCreationDialog` is true.

suffle commented on Resources/Private/JavaScript/media-module/src/components/TopBar/SortOrderSelector.tsx:
```diff
@@ -0,0 +1,92 @@
+
+    return (
+        <div className={classes.sortingState}>
+            <div className={classes.selectBox}>
```
The selectbox itself doesn't seem to respect classnames

Sebobo replied:
Hm it should. The component supports it. Maybe a JSS related issue. Will check.

Sebobo replied:
OK the `className` is just added to the inner div. Your solution is fine for now. To do it proper, one would have to use the `themr` api.

suffle commented:
I tweaked the responsive behaviour a little, maybe we should think about making the Topbar multiline and collapsible on small screens?

Sebobo commented:
Ok works great. Can you remove the compiled css & js from your commits. Then I would merge and optimise the responsive behaviour in general a bit more afterwards.

suffle commented:
I was wondering if the compiled files should be in there, but I guess thats just for the ci build? Removed them and fixed the tests I broke. Can you give me a little nudge, when you are done optimizing? Would love to see, what you did
